dm/syncer/optimist.go

```diff
@@ -163,16 +163,22 @@ func (s *Syncer) handleQueryEventOptimistic(qec *queryEventContext) error {
 		}
 	}
 
+	s.tctx.L().Info("putted a shard DDL info into etcd", zap.Stringer("info", info))
 	if !skipOp {
-		s.tctx.L().Info("putted a shard DDL info into etcd", zap.Stringer("info", info))
-		op, err = s.optimist.GetOperation(qec.tctx.Ctx, info, rev+1)
-		if err != nil {
-			return err
+		for {
+			op, err = s.optimist.GetOperation(qec.tctx.Ctx, info, rev+1)
+			if err != nil {
+				return err
+			}
+			s.tctx.L().Info("got a shard DDL lock operation", zap.Stringer("operation", op))
+			if op.ConflictStage != optimism.ConflictDetected {
+				break
+			}
+			rev = op.Revision
+			s.tctx.L().Info("operation conflict detected, waiting for resolve", zap.Stringer("info", info))
```

Comment:
Should add some backoff timeout here to avoid retry too fast. I think we can also replace this for loop by just return an error as the original logic, in this way, we can still depend on the auto-resume logic to trigger manually handle this error.

Reply:
We won't receive too many operations here, so I think without backoff is still okay.
For the second suggestion, the current implementation is more like the pessimistic lock. If after a discussion we think it's necessary to change this logic to auto-resume, we can do this in another PR which can also revise pessimistic lock's logic.

Reply:
I think return error will be better for user to notice, but it's difficult to handle it because the task is already been paused.